data-generation/generate-instructions.py

```diff
@@ -0,0 +1,204 @@
+from random import randint, random
+
+
+"""
+Classes
+"""
+
+
+class NoActionException(Exception):
+    pass
+
+
+class Block:
```

Comment:
Because the final result we are going for is really the trail of operations left behind by the "solving". If it flipped it in place then we would either have to clone each object before we added it to a new configuration in one step of the solving process, or we would be flipping blocks that were already placed in a configuration. I thought it made more sense to just have each operation return a new object rather than scatter clones all over our code.

Reply:
Okay, that sounds like a good idea in that case.